src/maestro/agents/code_agent.py

```diff
@@ -18,6 +22,88 @@ def __init__(self, agent: dict) -> None:
         Initializes the agent with agent definitions.
         """
         super().__init__(agent)
+        self.agent = agent  # Store the agent dictionary for accessing metadata
+
+    def _install_dependencies(self) -> None:
+        """
+        Check if the agent has dependencies in its metadata and install them if they exist.
+        """
+        dependencies = self.agent.get("metadata", {}).get("dependencies")
+        if not dependencies or dependencies.strip() == "":
+            return
+
+        self.print(f"Installing dependencies for {self.agent_name}...")
+
+        temp_file_path = None
+        try:
+            # Create a temporary requirements.txt file
+            with tempfile.NamedTemporaryFile(
+                mode="w", delete=False, suffix=".txt"
+            ) as temp_file:
+                temp_file_path = temp_file.name
+                temp_file.write(dependencies)
```

Comment:
```suggestion
                temp_file.write(get_content(dependencies))
```

In theory this would add support for file path (would need to import the utility function). Though it may make more sense to call the utility at a different place

Reply:
Think it actually required a second arg to point to the agent yaml, but added support for this.